src/token_contract/src/main.nr

```diff
@@ -1,45 +1,30 @@
 mod types;
 mod test;
 
+use crate::types::token_utils::token;
 use dep::aztec::macros::aztec;
 
-// Minimal token implementation that supports `AuthWit` accounts.
-// The auth message follows a similar pattern to the cross-chain message and includes a designated caller.
-// The designated caller is ALWAYS used here, and not based on a flag as cross-chain.
-// message hash = H([caller, contract, selector, ...args])
-// To be read as `caller` calls function at `contract` defined by `selector` with `args`
-// Including a nonce in the message hash ensures that the message can only be used once.
+#[token]
 #[aztec]
 pub contract Token {
 
     use dep::compressed_string::FieldCompressedString;
     use dep::std::ops::{Add, Sub};
 
     use dep::aztec::{
-        context::{PrivateCallInterface, PrivateContext},
         encrypted_logs::log_assembly_strategies::default_aes128::note::encode_and_encrypt_note,
         macros::{functions::{initializer, internal, private, public, view}, storage::storage},
-        note::{note_emission::OuterNoteEmission, note_interface::PartialNote},
-        oracle::random::random,
-        prelude::{AztecAddress, Map, PublicContext, PublicImmutable, PublicMutable},
+        prelude::{AztecAddress, Map, PublicImmutable, PublicMutable},
         protocol_types::{point::Point, traits::{Hash, Serialize}},
```

Comment:
w/o LSP well configured and working, these required imports are very hard to find